core/bignum/bignum_spec.rb

```diff
@@ -16,4 +16,10 @@
       Bignum.new
     end.should raise_error(NoMethodError)
   end
+
+  ruby_version_is '2.4' do
+    it "is deprecated and unified into Integer" do
+      Bignum.should == Integer
```

Comment:
It would interesting to test for the deprecation message, but in a separate example, with `-> { ... }.should complain(/constant ::Fixnum is deprecated/)`

Reply:
Could you use `equal` instead of `==` here to stress they refer to the same Class instance?

Reply:
@eregon I cannot catch deprecation warning in spec. Looks like it's magic

Reply:
It's a bit tricky, I'll do it.

Reply:
It was a MSpec bug, fairly confusing indeed: https://github.com/ruby/mspec/commit/b6065feb82f57d98780b2f8c247a36fd91b56f85